Dbertozzi/feature my shop page #15

dbertozzi opened the pull request:
Have some errors here but it's functional. Noted some of the details in most recent commit. I have to get to a thanksgiving dinner now.

dbertozzi committed:
Update from dev conflict resolution

dbertozzi committed:
Basic Look of Shop Page

Basic look of shop page with products. Need to fetch data still.

dbertozzi committed:
Merge branch 'dev' into dbertozzi/feature-my-shop-page-#15

dbertozzi committed:
Fetch API Initial Setup

-Need to add useReducer: "You can also replace multiple useState variables with useReducer if 'setCardLiked' needs the current value of 'shop.items.length'  react-hooks/exhaustive-deps"
-Navbar has a non-unique key error somewhere that needs to be fixed.
-the look of the page needs work still

naapak committed:
.env file deleted

naapak commented on client/src/pages/Shop.js:
```diff
@@ -0,0 +1,230 @@
+  const token = localStorage.getItem("token");
+  const USER_API = "http://localhost:3001/users/";
+  // user id will be from userContext in the future
+  const userid = "5da4c98f912176179cb35e43";
```
you can use the _id from the token. Use jwt.decode(token) to get the user details.

naapak reviewed (commented):
Get the user._id from the token. Apart from that looks good.

dbertozzi committed:
Added reducer to shop and got userid from token

State required multiple useState calls in succession and some relied on each other which was creating bugs. I added a reducer to solve this issue. I decoded the JWT Token to get the userid that was previously hard coded.

dbertozzi committed:
Merge branch 'dbertozzi/feature-my-shop-page-#15' of https://github.com/hatchways/team-chestnut into dbertozzi/feature-my-shop-page-#15
